react/homework.md

```diff
@@ -0,0 +1,135 @@
+8. Use the search page from the first part and rewrite it. Split it into 2 logical parts:
+   - **Search bar.** Enter text -> Press Enter -> Send the request to API with the searching parameters -> the list of results is updating
+   - **The result list of searching**: display the data that the API query returned using Cards.
+9. A customer decided to change requirement. Now every element of the list has to show a small piece of information. After the click on the element we have to show all the  available information about the element in the modal window. The modal window should be closed by clicking on the cross button in the upper right corner or by clicking on a page outside the modal window. When we open a modal window the page should be covered with overlay.
+10. Create a download indicator (a component with animation, or at least with the string "Downloading"). You should place this indicator with your own sense of beauty.
+
+All the logical parts must be in separated components.
+**User-friendly interface with the downloading indicator and with notification messages in a case if something goes wrong or some information cannot be found is warmly welcome.**
+
+11. Retain the functionality and **rewrite the form** with library **React Hook Form**. 
+12. Implement state management solution using useReducer and Context API for search results and form data saving.
+13. Add to global state
+    - **switches for sorting** (at least three sorting options). Select a parameter -> send a request to the API with the entered parameter -> the list of search results is updated
+    - **switches for pagination.** It should be possible to select the number of results per page and the number of the page we are currently on, the total number of pages. Select a parameter -> send a request to the API with the entered parameter -> the list of search results is updated
+14. The customer has changed the requirements for the output: 
+    - Now, when you click on each item from the search results, you do not need to open a modal, but open the information on the same div. 
+    - Also add a "Back" link on this div that should allow to hide the information and show the search bar with results again. 
+    - Display the current position in the header. 
+    - The detail page should take data from the global state.
+
+### Score
+
+The task will be checked by the mentor during the first 6 weeks. Send a codesandbox link to your mentor.
+
+1. Search bar + components – **2 points**
+2. Form + Cards - **2 points**
+3. API and modal window - **2 points**
+4. Custom State Management - **2 points**
+5. Sorting and pagination - **2 points**
+6. Search details - **2 points**
+
+If one of the items mentioned above is not accomplished – grade for the task is **0 points**. If there are direct DOM manipulations – **0 points**.
+
+### Theory
```

Comment:
please filter this list
we want to have only ENG sources

Reply:
done